### PR TITLE
Add Pipenv lockfile support (Resolves #71)

### LINKED ISSUE #71: Add support for pipenv lock files
Currently `Pipfile.lock` is not a support lock file, please add support.

### PR DESCRIPTION
**Pull request by newdominic**
Add support for `Pipenv.lock`. Fixes #71

### CODE REVIEW

**Comment by oliverchang** [comment]
Thank you very much for the contribution! CC @G-Rath as well who contributed most of the existing lockfile code.

**Comment by G-Rath** [comment]
@oliverchang @another-rex @newdominic would one of you mind adding "Resolves #71" to the PR description? (I've not the necessary permissions)

**Review by G-Rath — APPROVED** [review]
LGTM!

**Comment by inferno-chromium** [comment]
> @oliverchang @another-rex @newdominic would one of you mind adding "Resolves #71" to the PR description? (I've not the necessary permissions)

done! we should start thinking about adding more project contributors early next year. sorry for the hassle till then.

**Comment by G-Rath** [comment]
@inferno-chromium thanks, though you've added it to the PR title when it needs to be in the description for it to count as a link😅

**Comment by inferno-chromium** [comment]
> @inferno-chromium thanks, though you've added it to the PR title when it needs to be in the description for it to count as a link😅

Done, forgot that.

File: pkg/lockfile/ecosystems_test.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ func TestKnownEcosystems(t *testing.T) {
 
 	expectedCount := numberOfLockfileParsers(t)
 
-	// npm, yarn, and pnpm, and pip and poetry, and maven and gradle, all
+	// npm, yarn, and pnpm, and pip, poetry, and pipenv, and maven and gradle, all
```
Comment (another-rex) [inline pkg/lockfile/ecosystems_test.go:37]:
```suggestion
	// - npm, yarn, and pnpm,
	// - pip, poetry, and pipenv,
	// - maven and gradle, all
```

**Review by another-rex — COMMENTED** [review]
Thank you! Can you also add a test case with a package that doesn't have a version? This can happen if a package is a local path for example. 

e.g. This [lockfile entry](https://github.com/google/osv.dev/blob/0673d64a21209b820d45a247ca4c750d9fe08ee1/docker/worker/Pipfile.lock#L382-L384)

**Review by oliverchang — APPROVED** [review]
thank you very much! 

Sorry for the delay in review -- most of our team was out over the holiday period.

**Comment by oliverchang** [comment]
@newdominic Looks like there's a minor merge conflict. Would you mind fixing this up? I don't seem to be able to push to your fork branch to do this myself.

**Review by another-rex — APPROVED** [review]

**Comment by another-rex** [comment]
LGTM, thank you!

**Pull request merged**